confdb/confdb.go

```diff
@@ -1837,27 +1892,163 @@ func get(subKeys []string, index int, node map[string]json.RawMessage, result *a
 		}
 
 		if len(results) == 0 {
-			pathPrefix := strings.Join(subKeys[:index+1], ".")
 			return pathErrorf("no value was found under path %q", pathPrefix)
 		}
 
 		*result = results
 		return nil
 	}
 
-	// decode the next map level
-	var level map[string]json.RawMessage
-	if err := jsonutil.DecodeWithNumber(bytes.NewReader(rawLevel), &level); err != nil {
-		if uErr, ok := err.(*json.UnmarshalTypeError); ok {
-			pathPrefix := strings.Join(subKeys[:index+1], ".")
-			return fmt.Errorf("cannot read path prefix %q: prefix maps to %s", pathPrefix, uErr.Value)
-		}
+	level, err := unmarshalLevel(subKeys, index, rawLevel)
+	if err != nil {
 		return err
 	}
 
 	return get(subKeys, index+1, level, result)
 }
 
+// getList traverses node (a decoded JSON list) and, depending on the path being
+// followed, does one of the following:
+//   - decodes a value from it into the result parameter
+//   - decodes all list elements, if the path ends in an unmatched placeholder
+//   - goes into one specific sub-path and recurses into get()
+//   - goes into potentially many sub-paths and accumulates the results, if the
+//     current path sub-key is an unmatched placeholder
+func getList(subKeys []accessor, keyIndex int, list []json.RawMessage, result *any) error {
+	pathPrefix := joinAccessors(subKeys[:keyIndex+1])
+	key := subKeys[keyIndex]
+
+	var matchAll bool
+	listIndex := -1
+	if key.keyType() == listIndexType {
+		listIndex, _ = strconv.Atoi(key.name())
+	} else if key.keyType() == indexPlaceholderType {
+		matchAll = true
+	} else {
+		return fmt.Errorf("key %q cannot be used to index list at path %q", key, pathPrefix)
```

Comment:
I think you want this?
```suggestion
		return fmt.Errorf("key %q cannot be used to index list at path %q", key.access(), pathPrefix)
```